tests/test_measureApCorr.py

```diff
@@ -199,6 +207,49 @@ def testApertureMeasTens(self):
         struct.apCorrMap[apFluxName].fillImage(test_fill)
         np.testing.assert_allclose(test_fill.getArray(), default_fill.getArray())
 
+    def testFilterBadValue(self):
+        """Check that the aperture correction filters a bad value."""
+        sourceCat = self.makeCatalog()
+        source = sourceCat.addNew()
+        nameAp = self.names[0] + self.apNameStr
+        source[nameAp + "_instFlux"] = 100.0
+        source[nameAp + "_instFluxErr"] = 1e-3
+        source[self.meas_apCorr_task.config.refFluxName + "_instFlux"] = 5.1
+        source[self.meas_apCorr_task.config.refFluxName + "_instFluxErr"] = 1e-3
+        source[self.unresolvedName] = 0.0
+        centroidKey = afwTable.Point2DKey(self.schema["slot_Centroid"])
+        x = self.exposure.getX0() + 1
+        y = self.exposure.getY0() + 1
+        source.set(centroidKey, lsst.geom.Point2D(x, y))
+
+        self.meas_apCorr_task.run(catalog=sourceCat, exposure=self.exposure)
+
+        # Check that both Ap fluxes are removed as outliers; one is due
+        # to being unfilled (nan), the other is a large outlier.
+        for name in self.names:
+            apCorrFlagName = "apcorr_" + name + self.apNameStr + "_used"
+            # Check that all but the final source are used.
+            self.assertTrue(sourceCat[apCorrFlagName][0: -1].all())
+            # Check that the final source is not used.
+            self.assertFalse(sourceCat[apCorrFlagName][-1])
+
+    def testTooFewSourcesAfterFiltering(self):
+        """Check that the aperture correction fails when too many are filtered."""
+        sourceCat = self.makeCatalog()
+        self.meas_apCorr_task.config.minDegreesOfFreedom = 4
+
+        for name in self.names:
+            nameAp = name + self.apNameStr
+            sourceCat[nameAp + "_instFlux"][0] = 100.0
+
+        with self.assertRaisesRegex(RuntimeError, "only 4 sources remain"):
+            self.meas_apCorr_task.run(catalog=sourceCat, exposure=self.exposure)
+
+        # With the measurement algorithm declared as something that might fail, should not get an exception
```

Comment:
Something in this sentence seems awry.

Reply:
I copied it from another test, but I have rephrased both of them to be clearer.